packages/cli-e2e/__tests__/angular.specs.ts

```diff
@@ -13,92 +13,128 @@ import {isSearchRequest} from '../utils/platform';
-    const clientPort = '4200';
-    const projectName = `${process.env.GITHUB_ACTION}-angular-project`;
-    const searchPageEndpoint = `http://localhost:${clientPort}`;
-    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
-    let interceptedRequests: HTTPRequest[] = [];
-    let page: Page;
-
-    const searchboxSelector = 'app-search-page app-search-box input';
-
-    beforeAll(async () => {
-      processManager = new ProcessManager();
-      browser = await getNewBrowser();
-      const buildProcess = setupUIProject(
-        processManager,
-        'ui:create:angular',
-        projectName,
-        {
-          flags: ['--defaults'],
-        }
-      );
-
-      buildProcess.stdout.on('data', async (data) => {
-        if (isGenericYesNoPrompt(data.toString())) {
-          await answerPrompt(`y${EOL}`, buildProcess);
-        }
-      });
-
-      await Promise.race([
-        new Promise<void>((resolve) => {
-          buildProcess.on('exit', async () => {
-            resolve();
-          });
-        }),
-        new Promise<void>((resolve) => {
-          buildProcess.stdout.on('data', (data) => {
-            if (
-              /Happy hacking !/.test(
-                stripAnsi(data.toString()).replace(/\n/g, '')
-              )
-            ) {
-              resolve();
-            }
-          });
-        }),
-      ]);
-
-      const startServerProcess = processManager.spawn('npm', ['run', 'start'], {
-        cwd: getProjectPath(projectName),
-      });
-
-      await new Promise<void>((resolve) => {
-        startServerProcess.stdout.on('data', async (data) => {
+    return Promise.race([
+      new Promise<void>((resolve) => {
+        proc.on('exit', async () => {
+          resolve();
+        });
+      }),
+      new Promise<void>((resolve) => {
+        proc.stdout.on('data', (data) => {
           if (
-            /Compiled successfully/.test(
+            /Happy hacking !/.test(
               stripAnsi(data.toString()).replace(/\n/g, '')
             )
           ) {
             resolve();
           }
         });
+      }),
+    ]);
+  };
+
+  const startApplication = async (processManager: ProcessManager) => {
+    const proc = processManager.spawn('npm', ['run', 'start'], {
+      cwd: getProjectPath(projectName),
+    });
+
+    return new Promise<void>((resolve) => {
+      proc.stdout.on('data', async (data) => {
+        if (
+          /Compiled successfully/.test(
+            stripAnsi(data.toString()).replace(/\n/g, '')
+          )
+        ) {
+          resolve();
+        }
       });
-    }, 420e3);
+    });
+  };
+
+  const deactivateEnvironmentFile = () => {
+    const pathToEnv = getProjectPath(projectName);
+    renameSync(join(pathToEnv, '.env'), join(pathToEnv, '.env.disabled'));
+  };
+
+  const restoreEnvironmentFile = () => {
+    const pathToEnv = getProjectPath(projectName);
+    renameSync(join(pathToEnv, '.env.disabled'), join(pathToEnv, '.env'));
+  };
```

Comment:
Nice methods, can be refactored to be even neatier tho, see [that](https://www.typescriptlang.org/play?#code/PTAEFkHsGMGsEsB2BzU8C2AHSAnALqNAIaIDkBARgKagAWRAzgwDZUAmoA7vHraL0TwA6ALAAoNlWjMiOGtEiIGBOYiLoqAZQCeiaAC5QACknLDynEmQAaUADMckdObyWUASlABeAHygAbpDwbADc4pLSsvKKyqDIVHgACo4AVlJJgrSGRpip6QBy6lQubsievqAWVmESUjJyhDEEuZBp0HiFGiXV4XVRjUoEKUGI2ULjssgM3SgA2gC65X5VKOIgoACiiByQdvy0NOgwCKti4gqDoJJE7fD+guxb-gBi8KydNF6gpEJUiP5CNjwBhECisNikGoXWI3PB3KhPV6sNQabzfX7-SHic5NSqcIiYJ5oowOZyVVxWWx4SAzMrePwAb3EoBZA1imEyABVIESvvEknl2olMjlBR0iu4aqzQKoijo9EZhkgclyef9bKT3LYlYgVbxuU8qZB3JLxABfGo4y7XW73PAI-7wRyIDSIPBIz7GJZ4glPIyw+GIt5-Iq2G1wu2Pf4ej6ms5iaEqKjKXAO-xOxSu93B4nehj4wn-ExUAORthB96h0CltMelFUdxAA)

Reply:
Also, you should refactor that in an utils.